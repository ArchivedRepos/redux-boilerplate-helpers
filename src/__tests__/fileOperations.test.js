/* eslint-env jest */

import fs from 'fs';
import addReduxActions from '../fileOperations';
import {
  generateNames,
  setImports,
  createAction,
  createReducerCase,
  createConst,
} from '../createAction';

jest.mock('../createAction.js', () => ({
  generateNames: jest.fn(() => ({ camel: 'testName', constant: 'TEST_NAME' })),
  setImports: jest.fn(),
  createAction: jest.fn(),
  createReducerCase: jest.fn(),
  createConst: jest.fn(),
}));

jest.mock('fs', () => ({
  readdirSync: jest.fn(() => ['constants.js', 'actions.js', 'reducer.js']),
  readFileSync: jest.fn(() => ''),
  writeFileSync: jest.fn(),
}));

jest.mock('path', () => ({
  join: jest.fn(() => '/tmp'),
}));

describe('File Operations', () => {
  afterEach(() => {
    generateNames.mockClear();
    setImports.mockClear();
    createAction.mockClear();
    createReducerCase.mockClear();
    createConst.mockClear();
    fs.readFileSync.mockClear();
    fs.writeFileSync.mockClear();
    fs.readdirSync.mockClear();
  });

  it('adds an action to the files', () => {
    addReduxActions('./', []);

    expect(setImports.mock.calls).toMatchSnapshot();
    expect(createConst.mock.calls).toMatchSnapshot();
    expect(createAction.mock.calls).toMatchSnapshot();
    expect(createReducerCase.mock.calls).toMatchSnapshot();
    expect(fs.writeFileSync.mock.calls).toMatchSnapshot();
    expect(fs.readFileSync.mock.calls).toMatchSnapshot();
    expect(fs.readdirSync.mock.calls).toMatchSnapshot();
  });

  it('adds multiple actions to the files', () => {
    const generateNamesOrig = require.requireActual('../createAction').generateNames;
    generateNames.mockImplementationOnce(generateNamesOrig);
    addReduxActions('./', ['testOne', 'testTwo', 'testThree']);

    expect(setImports.mock.calls).toMatchSnapshot();
    expect(createConst.mock.calls).toMatchSnapshot();
    expect(createAction.mock.calls).toMatchSnapshot();
    expect(createReducerCase.mock.calls).toMatchSnapshot();
    expect(fs.writeFileSync.mock.calls).toMatchSnapshot();
    expect(fs.readFileSync.mock.calls).toMatchSnapshot();
    expect(fs.readdirSync.mock.calls).toMatchSnapshot();
  });

  it('creates files that are not preset', () => {
    setImports.mockReset();
    fs.readdirSync.mockImplementationOnce(() => []);
    addReduxActions('./', []);

    expect(setImports.mock.calls).toMatchSnapshot();
    expect(createConst.mock.calls).toMatchSnapshot();
    expect(createAction.mock.calls).toMatchSnapshot();
    expect(createReducerCase.mock.calls).toMatchSnapshot();
    expect(fs.writeFileSync.mock.calls).toMatchSnapshot();
    expect(fs.readFileSync.mock.calls).toMatchSnapshot();
    expect(fs.readdirSync.mock.calls).toMatchSnapshot();
  });

  it('does a dry run', () => {
    const log = console.log;
    console.log = jest.fn();

    addReduxActions('./', [], { dry: true });
    console.log = log;
  });
});
