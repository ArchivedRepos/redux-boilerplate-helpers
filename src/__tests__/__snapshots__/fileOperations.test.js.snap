// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Operations adds an action to the files 1`] = `Array []`;

exports[`File Operations adds an action to the files 2`] = `Array []`;

exports[`File Operations adds an action to the files 3`] = `Array []`;

exports[`File Operations adds an action to the files 4`] = `Array []`;

exports[`File Operations adds an action to the files 5`] = `
Array [
  Array [
    "/tmp",
    "",
  ],
  Array [
    "/tmp",
    "",
  ],
  Array [
    "/tmp",
    "",
  ],
]
`;

exports[`File Operations adds an action to the files 6`] = `
Array [
  Array [
    "/tmp",
  ],
  Array [
    "/tmp",
  ],
  Array [
    "/tmp",
  ],
]
`;

exports[`File Operations adds an action to the files 7`] = `
Array [
  Array [
    "/tmp",
  ],
]
`;

exports[`File Operations adds multiple actions to the files 1`] = `
Array [
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testOne",
      "constant": "TEST_ONE",
    },
    Object {
      "insertHelper": true,
    },
  ],
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testOne",
      "constant": "TEST_ONE",
    },
  ],
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testName",
      "constant": "TEST_NAME",
    },
    Object {
      "insertHelper": true,
    },
  ],
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testName",
      "constant": "TEST_NAME",
    },
  ],
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testName",
      "constant": "TEST_NAME",
    },
    Object {
      "insertHelper": true,
    },
  ],
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testName",
      "constant": "TEST_NAME",
    },
  ],
]
`;

exports[`File Operations adds multiple actions to the files 2`] = `
Array [
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testOne",
      "constant": "TEST_ONE",
    },
    "/tmp",
  ],
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testName",
      "constant": "TEST_NAME",
    },
    "/tmp",
  ],
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testName",
      "constant": "TEST_NAME",
    },
    "/tmp",
  ],
]
`;

exports[`File Operations adds multiple actions to the files 3`] = `
Array [
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testOne",
      "constant": "TEST_ONE",
    },
  ],
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testName",
      "constant": "TEST_NAME",
    },
  ],
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testName",
      "constant": "TEST_NAME",
    },
  ],
]
`;

exports[`File Operations adds multiple actions to the files 4`] = `
Array [
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testOne",
      "constant": "TEST_ONE",
    },
  ],
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testName",
      "constant": "TEST_NAME",
    },
  ],
  Array [
    Object {
      "comments": null,
      "loc": Object {
        "end": Object {
          "column": 0,
          "line": 1,
        },
        "indent": 0,
        "lines": Lines {},
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": null,
      "needsParens": false,
      "program": Program {
        "body": Array [],
        "errors": Array [],
        "loc": Object {
          "end": Object {
            "column": 0,
            "line": 1,
          },
          "indent": 0,
          "lines": Lines {},
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "needsParens": false,
        "sourceType": "module",
        "type": "Program",
      },
      "type": "File",
    },
    Object {
      "camel": "testName",
      "constant": "TEST_NAME",
    },
  ],
]
`;

exports[`File Operations adds multiple actions to the files 5`] = `
Array [
  Array [
    "/tmp",
    "",
  ],
  Array [
    "/tmp",
    "",
  ],
  Array [
    "/tmp",
    "",
  ],
]
`;

exports[`File Operations adds multiple actions to the files 6`] = `
Array [
  Array [
    "/tmp",
  ],
  Array [
    "/tmp",
  ],
  Array [
    "/tmp",
  ],
]
`;

exports[`File Operations adds multiple actions to the files 7`] = `
Array [
  Array [
    "/tmp",
  ],
]
`;

exports[`File Operations creates files that are not preset 1`] = `Array []`;

exports[`File Operations creates files that are not preset 2`] = `Array []`;

exports[`File Operations creates files that are not preset 3`] = `Array []`;

exports[`File Operations creates files that are not preset 4`] = `Array []`;

exports[`File Operations creates files that are not preset 5`] = `
Array [
  Array [
    "/tmp",
    "",
  ],
  Array [
    "/tmp",
    "",
  ],
  Array [
    "/tmp",
    "",
  ],
  Array [
    "/tmp",
    "",
  ],
  Array [
    "/tmp",
    "",
  ],
  Array [
    "/tmp",
    "",
  ],
]
`;

exports[`File Operations creates files that are not preset 6`] = `
Array [
  Array [
    "../assets/reducer.js",
  ],
  Array [
    "/tmp",
  ],
  Array [
    "/tmp",
  ],
  Array [
    "/tmp",
  ],
]
`;

exports[`File Operations creates files that are not preset 7`] = `
Array [
  Array [
    "/tmp",
  ],
]
`;
